src/ansys/motorcad/core/methods/adaptive_geometry.py

```diff
@@ -0,0 +1,168 @@
+"""Methods for adaptive geometry."""
+from ansys.motorcad.core.geometry import (
+    Region,
+    _convert_entities_from_json,
+    _convert_entities_to_json,
+)
+
+
+class _RpcMethodsAdaptiveGeometry:
+    def __init__(self, mc_connection):
+        self.connection = mc_connection
+
+    def set_adaptive_parameter_value(self, name, value):
+        """Set adaptive parameter, if parameter does not exist then add it.
+
+        Parameters
+        ----------
+        name : string
+            name of parameter.
+        value : float
+            value of parameter.
+        """
+        self.connection.ensure_version_at_least("2024.0")
+        method = "SetAdaptiveParameterValue"
+        params = [name, value]
+        return self.connection.send_and_receive(method, params)
+
+    def get_adaptive_parameter_value(self, name):
+        """Get adaptive parameter.
+
+        Parameters
+        ----------
+        name : string
+            name of parameter.
+
+        Returns
+        -------
+        float
+            value of parameter.
+        """
+        self.connection.ensure_version_at_least("2024.0")
+        method = "GetAdaptiveParameterValue"
+        params = [name]
+        return self.connection.send_and_receive(method, params)
+
+    def get_region(self, name):
+        """Get Motor-CAD geometry region.
+
+        Parameters
+        ----------
+        name : string
+            name of region.
+
+        Returns
+        -------
+        ansys.motorcad.core.geometry.Region
+            Motor-CAD region object.
+
+        """
+        self.connection.ensure_version_at_least("2024.0")
+        method = "GetRegion"
+        params = [name]
+        raw_region = self.connection.send_and_receive(method, params)
+
+        region = Region()
+        region._from_json(raw_region)
+
+        return region
+
+    def set_region(self, region):
+        """Set Motor-CAD geometry region.
+
+        Parameters
+        ----------
+        region : ansys.motorcad.core.geometry.Region
+            Motor-CAD region object.
+        """
+        self.connection.ensure_version_at_least("2024.0")
+        raw_region = region._to_json()
+
+        method = "SetRegion"
+        params = [raw_region]
+        return self.connection.send_and_receive(method, params)
+
+    def get_entities_between_poly_start_end(self, region, polyline):
+        """Return entities from region which are between start/end of polyline.
+
+        Parameters
+        ----------
+        region : ansys.motorcad.core.geometry.Region
+            Motor-CAD region object.
+        polyline : List of Line/Arc objects
+            Polyline using lines/arcs
+
+        Returns
+        -------
+        :Object List
+            List of Line/Arc objects
+        """
+        self.connection.ensure_version_at_least("2024.0")
+        raw_region = region._to_json()
+        raw_entities = _convert_entities_to_json(polyline)
```

Comment:
```_convert_entities_to_json``` assumes we always pass a list. If we just pass a single line/arc this fails. Should be handled in ```_convert_entities_to_json```